jhub_apps/hub_client/hub_client.py

```diff
@@ -298,17 +301,36 @@ def get_groups(self):
         r.raise_for_status()
         return r.json()
 
+    @requires_user_token
+    def get_user_scopes(self):
+        assert self.token_json
+        assert self.token_json.get("scopes")
```

Comment:
Is it intended to raise when scopes are not in the dictionary, or if scopes are empty?

Reply:
>Is it intended to raise when scopes are not in the dictionary, or if scopes are empty?

This is intended for when scopes are not in dictionary, this is for when if jupyterhub API changes or unauthorized requests

Reply:
Then would it be better to test for:

```
assert "scopes" in self.token_json
```

Though maybe a user without any scopes is a bit pathological edge case?

Reply:
> Though maybe a user without any scopes is a bit pathological edge case?

Yep, it's very unlikely unless API changes.